src/ixdat/plotters/tpms_plotter.py

```diff
@@ -146,139 +141,452 @@ def plot_measurement(
                 **kwargs,
             )
 
-        # Then we have meta data plottet using MSPlotter
-        self.ms_plotter.plot_measurement(
-            measurement=measurement,
-            axes=[axes[1], axes[3]] if TP_lists else [axes[1]],
-            tspan=tspan,
-            mass_list=TP_list,
-            mass_lists=TP_lists,
-            unit=None,
-            x_unit=x_unit,
-            logplot=False,
-        )
+        T_name = T_name or measurement.T_name
+        P_name = P_name or measurement.P_name
 
-        # Set correct ylabel and overwrite colours if colours is manually set
-        axes[1].set_ylabel(T_name)
-        if not T_color:
-            T_color = axes[1].get_lines()[0].get_color()
-        else:
-            axes[1].get_lines()[0].set_color(T_color)
-        color_axis(axes[1], T_color)
+        if not meta_lists:
```

Comment:
This whole *meta_list(s)* thing can simplify, I think, by instead using the more transparent `T_name_list` and `P_name_list` idea